pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -594,6 +604,28 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 				Value: string(vmi.Spec.Domain.Firmware.UUID),
 			},
 		}
+
+		if vmi.Spec.Domain.Firmware.Bootloader != nil && vmi.Spec.Domain.Firmware.Bootloader.EFI != nil {
+
+			var efi string
+
+			if vmi.Spec.Domain.Firmware.Bootloader.EFI.Secure {
+				efi = EFISecurePath
+			} else {
+				efi = EFIPath
+			}
+			domain.Spec.OS.BootLoader = &Loader{
+				Path:     efi,
+				ReadOnly: "yes",
+				Secure:   boolToYesNo(&vmi.Spec.Domain.Firmware.Bootloader.EFI.Secure, false),
+				Type:     "pflash",
+			}
+
+			domain.Spec.OS.NVRam = &NVRam{
```

Comment:
I believe we confuse `nvram` and `template` parameters, I checked the configuration under the oVirt and it looks like
```
  <os>
    <type arch='x86_64' machine='pc-q35-rhel7.6.0'>hvm</type>
    <loader readonly='yes' secure='yes' type='pflash'>/usr/share/OVMF/OVMF_CODE.secboot.fd</loader>
    <nvram template='/usr/share/OVMF/OVMF_VARS.secboot.fd'>/var/lib/libvirt/qemu/nvram/1c08a8a2-3bc9-41d8-aafc-537e9d572e49.fd</nvram>
    <smbios mode='sysinfo'/>
  </os>
```
and the only reason why it works for us, it because libvirt has default mapping.

I think it must be something like
```
...
EFIVarsSecurePath      = "/usr/share/OVMF/OVMF_VARS.secboot.fd"
...
		if vmi.Spec.Domain.Firmware.Bootloader != nil && vmi.Spec.Domain.Firmware.Bootloader.EFI != nil {

			var efi string
			var nvramTemplate string
			if vmi.Spec.Domain.Firmware.Bootloader.EFI.Secure {
				efi = EFISecurePath
				nvramTemplate = EFIVarsSecurePath
			} else {
				efi = EFIPath
				nvramTemplate = EFIVarsPath
			}
			domain.Spec.OS.BootLoader = &Loader{
				Path:     efi,
				ReadOnly: "yes",
				Secure:   boolToYesNo(&vmi.Spec.Domain.Firmware.Bootloader.EFI.Secure, false),
				Type:     "pflash",
			}

			domain.Spec.OS.NVRam = &NVRam{
				NVRam:    fmt.Sprintf("/tmp/%s_VARS.fd", domain.Spec.Name),
				Template: nvramTemplate,
			}
		}
```

Reply:
:+1: great catch @cynepco3hahue You're right. 

I'll fix this. We got lucky that our current configuration just happens to work.

Reply:
@cynepco3hahue 

I can't get secure UEFI to boot with the usr/share/OVMF/OVMF_VARS.secboot.fd template. It will only boot with the usr/share/OVMF/OVMF_VARS.fd template for some reason.  When I use the secure vars the bootloader says there's "no boot device" 

any ideas why this might be occurring?